go/worker/keymanager/worker.go

```diff
@@ -809,11 +902,50 @@ func (w *Worker) randomBlockHeight(epoch beacon.EpochTime, percentile int64) (in
 	return height, nil
 }
 
+func (w *Worker) updateGenerateMasterSecretEpoch() {
+	var nextEpoch beacon.EpochTime
+
+	// If at least one master secret has been generated, respect the rotation interval.
+	nextGen := w.kmStatus.NextGeneration()
+	if nextGen != 0 {
+		// Disable rotation if the policy is not set.
+		var rotationInterval beacon.EpochTime
+		if w.kmStatus.Policy != nil {
+			rotationInterval = w.kmStatus.Policy.Policy.MasterSecretRotationInterval
+		}
+
+		// Secrets are allowed to be generated at most one epoch before the rotation.
+		nextEpoch = w.kmStatus.RotationEpoch + rotationInterval - 1
+
+		// Rotation not allowed.
+		if rotationInterval == 0 {
+			nextEpoch = math.MaxUint64
+		}
```

Comment:
Cannot move to the `else` section as the policy can also set the rotation interval to 0 (rotations disabled).
Cannot initialize to `math.MaxUint64` because if no secrets have been proposed yet, `nextEpoch` should be 0 (meaning we should generate the secret immediately).